src/ray/common/asio/postable.h

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include "ray/common/asio/instrumented_io_context.h"
+#include "ray/util/function_traits.h"
+
+namespace ray {
+
+template <typename FuncType>
+class Postable;
+
+namespace internal {
+
+template <typename FuncType>
+struct ToPostableHelper;
+
+template <typename FuncType>
+struct ToPostableHelper<std::function<FuncType>> {
+  using type = Postable<FuncType>;
+};
+
+}  // namespace internal
+
+template <typename FuncType>
+using ToPostable = typename internal::ToPostableHelper<FuncType>::type;
+
+template <typename FuncType>
+class Postable {
+ public:
+  Postable(std::function<FuncType> func, instrumented_io_context &io_context)
+      : func_(std::move(func)), io_context_(io_context) {}
+
+  template <typename... Args>
+  void Post(const std::string &name, Args &&...args) const {
```

Comment:
Thinking more: I think we can upgrade it a bit: support both Post and Dispatch. For in-memory KV store, we use Post and for Redis KV store, we use Dispatch.

Reply:
good idea, will do